src/m365/spfx/commands/project/project-github-workflow-add.ts

```diff
@@ -0,0 +1,195 @@
+            return isValidSharePointUrl;
+          }
+        }
+
+        if (args.options.loginMethod && SpfxProjectGithubWorkflowAddCommand.loginMethod.indexOf(args.options.loginMethod) < 0) {
+          return `${args.options.loginMethod} is not a valid login method. Allowed values are ${SpfxProjectGithubWorkflowAddCommand.loginMethod.join(', ')}`;
+        }
+
+        if (args.options.scope && SpfxProjectGithubWorkflowAddCommand.scope.indexOf(args.options.scope) < 0) {
+          return `${args.options.scope} is not a valid scope. Allowed values are ${SpfxProjectGithubWorkflowAddCommand.scope.join(', ')}`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    this.projectRootPath = this.getProjectRoot(process.cwd());
+    if (this.projectRootPath === null) {
+      throw new CommandError(`Couldn't find project root folder`, SpfxProjectGithubWorkflowAddCommand.ERROR_NO_PROJECT_ROOT_FOLDER);
+    }
+
+    const solutionPackageJsonFile: string = path.join(this.projectRootPath, 'package.json');
+    const packageJson: string = fs.readFileSync(solutionPackageJsonFile, 'utf-8');
+    const solutionName = JSON.parse(packageJson).name;
+
+    if (this.debug) {
+      logger.logToStderr(`Adding GitHub workflow in the current SPFx project`);
+    }
+
+    try {
+      this.updateWorkflow(solutionName, workflow, args.options);
+      this.saveWorkflow(workflow);
+    }
+    catch (error: any) {
+      throw new CommandError(error);
+    }
+  }
+
+  private saveWorkflow(workflow: gitHubWorkflow): void {
+    const githubPath: string = path.join(this.projectRootPath as string, '.github');
+    fsUtil.ensureDirectory(githubPath);
+
+    const workflowPath: string = path.join(githubPath, 'workflows');
+    fsUtil.ensureDirectory(workflowPath);
+
+    const workflowFile: string = path.join(workflowPath, 'deploy-spfx-solution.yml');
+    fs.writeFileSync(path.resolve(workflowFile), yaml.stringify(workflow), 'utf-8');
+  }
+
+  private updateWorkflow(solutionName: string, workflow: gitHubWorkflow, options: GlobalOptions): void {
+    workflow.name = workflow.name.replace('{{ name }}', options.name ?? solutionName);
+
+    if (options.branchName) {
+      workflow.on.push.branches[0] = options.branchName;
+    }
+
+    if (options.manuallyTrigger) {
+      // eslint-disable-next-line camelcase
+      workflow.on.workflow_dispatch = null;
+    }
+
+    if (options.skipFeatureDeployment) {
+      this.getDeployAction(workflow).with.SKIP_FEATURE_DEPLOYMENT = true;
+    }
+
+    if (options.overwrite) {
+      this.getDeployAction(workflow).with.OVERWRITE = true;
+    }
+
+    if (options.loginMethod === 'user') {
+      const loginAction = this.getLoginAction(workflow);
+      loginAction.with = {};
+      loginAction.with.ADMIN_USERNAME = '${{ secrets.ADMIN_USERNAME }}';
+      loginAction.with.ADMIN_PASSWORD = '${{ secrets.ADMIN_PASSWORD }}';
```

Comment:
We can simplify this and assign all properties at once, rather than in 3 separate statements

Reply:
✅